### PR TITLE
Extend checkstyle test to validate license file

### PR DESCRIPTION
**Pull request by dmitrii-ubskii**
## What is the goal of this PR?

We allow checkstyle to validate the license files.

## What are the changes implemented in this PR?

1. `checkstyle_test()` no longer has an implicit ignore list;
2. each kind of license now has a header and a full text template in tool/checkstyle/config;
3. checkstyle.py wrapper removed — it was just acting as a shell script anyway;
5. cleaned up the LICENSE.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -617,45 +617,3 @@ Program, unless a warranty or assumption of liability accompanies a
...
-get its source.  For example, if your program is a web application, its
-interface could display a "Source" link that leads users to an archive
-of the code.  There are many ways you could offer source, and different
-solutions will be better for different programs; see section 13 for the
-specific requirements.
-
-  You should also get your employer (if you work as a programmer) or school,
-if any, to sign a "copyright disclaimer" for the program, if necessary.
-For more information on this, and how to apply and follow the GNU AGPL, see
-<https://www.gnu.org/licenses/>.
```
Comment (dmitrii-ubskii) [inline LICENSE:661]: This is the example header for the source files, it should not be here.

File: tool/checkstyle/config/BUILD [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
     srcs = [
-        "checkstyle-file-header-agpl.txt",
-        "checkstyle-file-header-apache.txt",
-        "checkstyle-file-header-commercial.txt",
+        "checkstyle-file-agpl-header.txt",
+        "checkstyle-file-apache-header.txt",
+        "checkstyle-file-commercial-header.txt",
+        "checkstyle-file-agpl-fulltext.txt",
+        "checkstyle-file-apache-fulltext.txt",
+        "checkstyle-file-commercial-fulltext.txt",
```
Comment (dmitrii-ubskii) [inline tool/checkstyle/config/BUILD:9]: The header pattern files were renamed to correspond to the values of `license_type` in `checkstyle_test()`

File: tool/checkstyle/rules.bzl [hunk 3]
```diff
@@ -145,6 +144,17 @@ checkstyle_test = rule(
         ),
     },
     outputs = {
-        "checkstyle_script": "%{name}.py",
+        "checkstyle_wrapper": "%{name}.kt",
```
Comment (dmitrii-ubskii) [inline tool/checkstyle/rules.bzl:147]: Technically not a script anymore.

File: tool/checkstyle/rules.bzl [hunk 4]
```diff
@@ -67,33 +64,39 @@ def _checkstyle_test_impl(ctx):
...
     )
     return DefaultInfo(
-        executable = ctx.outputs.checkstyle_script,
+        executable = ctx.outputs.checkstyle_wrapper,
         files = depset(files),
         runfiles = runfiles,
     )
 
-checkstyle_test = rule(
+_checkstyle_test = rule(
```
Comment (dmitrii-ubskii) [inline tool/checkstyle/rules.bzl:86]: Another layer of indirection. This rule will later be wrapped in a `kt_jvm_test()` in the real `checkstyle_test()`.

File: tool/checkstyle/rules.bzl [hunk 5]
```diff
@@ -66,34 +62,39 @@ def _checkstyle_test_impl(ctx):
...
-        files = depset(files),
+        executable = checkstyle_wrapper,
         runfiles = runfiles,
     )
 
-checkstyle_test = rule(
-    implementation = _checkstyle_test_impl,
-    test = True,
+_checkstyle_wrapper = rule(
+    implementation = _checkstyle_wrapper_impl,
```
Comment (dmitrii-ubskii) [inline tool/checkstyle/rules.bzl:86]: This rule can no longer be run as a test—it needs to be compiled as a `kt_jvm_test`—, so it no longer gets the suffix.

File: library/rocksdbjni/BuilderLinux.kt [hunk 6]
```diff
@@ -2,6 +2,8 @@ package com.vaticle.dependencies.library.rocksdbjni
 
 import java.nio.file.Paths
 
+import com.vaticle.dependencies.util.bash
```
Comment (flyingsilverfin) [inline library/rocksdbjni/BuilderLinux.kt:5]: unused import?

File: tool/checkstyle/rules.bzl [hunk 7]
```diff
@@ -66,23 +60,20 @@ def _checkstyle_test_impl(ctx):
...
-        substitutions = {
-            "{command}" : cmd,
-            "{allow_failure}": str(int(ctx.attr.allow_failure)),
-        },
+    checkstyle_wrapper = ctx.actions.declare_file("%s.sh" % ctx.attr.name)
+    ctx.actions.write(
+        output = checkstyle_wrapper,
+        content = cmd,
         is_executable = True,
     )
```
Comment (dmitrii-ubskii) [inline tool/checkstyle/rules.bzl:68]: We now write the command directly to a shell script, with no intermediaries.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**